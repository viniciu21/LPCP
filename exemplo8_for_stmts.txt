declaration:
    a: int;
    b: int;
    c: int;
    x: int;
end_declaration

main:
    for(x = 1; x < 10; x = x + 1):
        a = 5;
    end_for;
end_main