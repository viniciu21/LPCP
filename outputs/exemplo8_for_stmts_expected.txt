Declaração de variável: Id "a" (2,5)(True,[(Id "a" (2,5),IntValue 0 (2,8))],[],[],[])
Declaração de variável: Id "x" (3,5)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "x" (3,5),IntValue 0 (3,8))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,9)(True,[(Id "a" (2,5),IntValue 0 (2,8)),(Id "x" (3,5),IntValue 0 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 5 (2,8)),(Id "x" (3,5),IntValue 0 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 5 (2,8)),(Id "x" (3,5),IntValue 1 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 10 (2,8)),(Id "x" (3,5),IntValue 1 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 10 (2,8)),(Id "x" (3,5),IntValue 2 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 15 (2,8)),(Id "x" (3,5),IntValue 2 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 15 (2,8)),(Id "x" (3,5),IntValue 3 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 20 (2,8)),(Id "x" (3,5),IntValue 3 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 20 (2,8)),(Id "x" (3,5),IntValue 4 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 25 (2,8)),(Id "x" (3,5),IntValue 4 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 25 (2,8)),(Id "x" (3,5),IntValue 5 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 30 (2,8)),(Id "x" (3,5),IntValue 5 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 30 (2,8)),(Id "x" (3,5),IntValue 6 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 35 (2,8)),(Id "x" (3,5),IntValue 6 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 35 (2,8)),(Id "x" (3,5),IntValue 7 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 40 (2,8)),(Id "x" (3,5),IntValue 7 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 40 (2,8)),(Id "x" (3,5),IntValue 8 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 45 (2,8)),(Id "x" (3,5),IntValue 8 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 45 (2,8)),(Id "x" (3,5),IntValue 9 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "a" (8,9)(True,[(Id "a" (2,5),IntValue 50 (2,8)),(Id "x" (3,5),IntValue 9 (7,13))],[],[],[])
Atualização de estado sobre a variável: Id "x" (7,24)(True,[(Id "a" (2,5),IntValue 50 (2,8)),(Id "x" (3,5),IntValue 10 (7,13))],[],[],[])
Tokens do programa: [Declaration (1,1),Colon (1,12),Id "a" (2,5),Colon (2,6),Type "int" (2,8),SemiColon (2,11),Id "x" (3,5),Colon (3,6),Type "int" (3,8),SemiColon (3,11),EndDeclaration (4,1),Main (6,1),Colon (6,5),For (7,5),Colon (7,34),Id "a" (8,9),Assign (8,11),Id "a" (8,13),Plus (8,15),IntValue 5 (8,17),SemiColon (8,18),EndFor (9,5),SemiColon (9,12),EndMain (10,1)]